modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsRepositoryImpl.java

```diff
@@ -46,26 +55,38 @@
      *
      * @param storeData If {@code true} - node stores data locally, {@code false} - otherwise.
      * @param persistence If {@code true} - node have persistence store, {@code false} - otherwise.
+     * @param subscriptionProcessor Subscription processor.
      * @param statisticsManager Ignite statistics manager.
      * @param log Ignite logger to use.
      */
     public IgniteStatisticsRepositoryImpl(
             boolean storeData,
             boolean persistence,
+            IgniteCacheDatabaseSharedManager database,
+            GridInternalSubscriptionProcessor subscriptionProcessor,
             IgniteStatisticsManagerImpl statisticsManager,
-            IgniteLogger log) {
+            Function<Class<?>, IgniteLogger> logSupplier
+    ) {
         if (storeData) {
             // Persistence store
-            partsStats = (persistence) ? null : new ConcurrentHashMap<>();
+            if (persistence) {
```

Comment:
whether the store should be persistent or not could be defined by presence of database

Reply:
Is is possible to split implementation for 2 particular cases: persistent and non-persistent.

Reply:
Done